Move sharing options into popover menu

ChristophWurst opened the pull request:
![bildschirmfoto von 2016-07-27 15-31-44](https://cloud.githubusercontent.com/assets/1374172/17176882/07c5aba0-540f-11e6-9535-8a98d17e823e.png)

@jancborchardt your turn, fix the styling :-P

TODO:
- [x] 'Unshare' button does not work. The event is not propagated to the registered event handler, no clue why cc @nextcloud/javascript 
- [x] with files, the 'can edit' checkbox can only be unchecked. If it's set again then some logic unsets it magically :weary: 

cc @nextcloud/designers

ChristophWurst committed:
move sharing options into popover menu

schiessle commented:
having this new drop-down menu for the additional share permissions looks great. But I would keep the "unshare" button out. This makes it way more easier to unshare multiple files in a row.

rullzer commented:
Awesome!
But I agree with @schiessle for the unshare button.

rullzer commented:
Btw why is can edit not in the dropdown?

ChristophWurst commented:
> Btw why is can edit not in the dropdown?

Because it's used more frequently according to @jancborchardt

MorrisJobke commented:
> Because it's used more frequently according to @jancborchardt

And it is a meta permission. `Can edit` includes all the other ones, right?

rullzer commented:
can edit does not cover can share...

jancborchardt commented:
»can edit« is outside of the dropdown because deciding if you share something with a person read-only or with full access is arguably an important decision. It should not be hidden away in a dropdown.

Agree that we can keep the »Unshare« button out of the dropdown for now. So it should look like this:

```
😎 Hans           [ ] can edit  •••  🗑

😃 Maria          [x] can edit  •••  🗑
```

jancborchardt commented:
What we should also do:
- [ ] Sort people with most permissions up top (those with edit rights)
- [ ] rename »can share« to »can reshare« so it’s more clear

ChristophWurst committed:
move unshare button out of popover menu; keep menu opened on re-render

rullzer commented:
I think the can reshare should also be out of the dropdown. It is an essential permission.

rullzer commented:
Ah it is because of some of the js magic. Let me fix it :)

rullzer committed:
Make can edit checkbox work for file shares

rullzer commented:
Should work now ;)

Some things:

- [ ] Can we move the ... and trash icon all the way to the right?
- [ ] Can we make the menu appear on the dots?

rullzer committed:
can share => can reshare

rullzer committed:
Move options to the right

rullzer commented:
![sharemenu](https://cloud.githubusercontent.com/assets/45821/17808081/3f52111a-660e-11e6-8492-04a9b11111f5.png)

It now looks like this. Which I think is very nice!
Only thing I really dislike here is the trashbin next to the dot menu... way to easy to click on it....

@jancborchardt suggestions?

MorrisJobke commented:
> Only thing I really dislike here is the trashbin next to the dot menu... way to easy to click on it....

And the alignment is a bit off.

rullzer commented:
From https://github.com/nextcloud/server/issues/146

- [ ] Rename `edit` to `upload and edit`

rullzer commented:
@MorrisJobke yeah I leave all that to somebody with a bit more CSS fu ;)

jancborchardt commented:
> Only thing I really dislike here is the trashbin next to the dot menu... way to easy to click on it....

@rullzer ;D exactly the reason why I would have moved it into the menu. ;) But a bit of spacing might take care of it as well.

MorrisJobke commented:
@ChristophWurst @rullzer What is the status of this? Could I somehow help here?

rullzer commented:
@MorrisJobke ah yes. The aligning is still a bit off and we probabaly need more spacing between the delete button.

Awesome if you could take care :)

ChristophWurst commented:
@MorrisJobke @rullzer what's the status of this now? Anything left that needs to be done?

MorrisJobke commented:
> @MorrisJobke @rullzer what's the status of this now? Anything left that needs to be done?

Rullzer asked if I could help with the layout. I haven't looked at it yet.

ChristophWurst commented:
Fixed the alignment and added some padding to the unshare icon:
![bildschirmfoto am 2016-10-03 um 09 34 43](https://cloud.githubusercontent.com/assets/1374172/19030238/aafc7822-894c-11e6-8595-42e5928a391d.png)

ChristophWurst committed:
fix sharing icon's alignment and add padding to the unshare icon

rullzer commented:
@ChristophWurst thnx!

I'd say lets get this in! We can sort out details later. Longer we wait the more conflicts and stuff will pop up.

SO LGTM!

LukasReschke commented:
LGTM

LukasReschke commented:
@ChristophWurst FYI https://github.com/nextcloud/server/blob/master/CONTRIBUTING.md#sign-your-work
